### ISSUE TITLE
Cpu consumed by `CheckStreamTimeoutLoop` is growing increasingly.

### ISSUE THREAD

**Issue by SunRunAway** [pingcap/tidb#14207]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

The CPU usage of a tidb-server with very low QPS would grow increasingly.

![image](https://user-images.githubusercontent.com/867381/71408861-94a61780-267a-11ea-9f94-5a030172add7.png)

![image](https://user-images.githubusercontent.com/867381/71408473-5ceaa000-2679-11ea-8827-c98f2addda5f.png)

```
[sunrunaway:~/Downloads]$ cat goroutine | grep CheckStreamTimeoutLoop | wc -l
7604
```

2. What did you expect to see?

3. What did you see instead?

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

v3.0.7

**Comment by SunRunAway** [comment]
The root cause is that `recycleIdleConnArray` does not stop its `CheckStreamTimeoutLoop`.

**Comment by SunRunAway** [comment]
We should fix it in v3.0.8

**Comment by SunRunAway** [comment]
This issue is fixed by https://github.com/pingcap/tidb/pull/13812, but not cherry-picked.
